arch/risc-v/include/irq.h

```diff
@@ -703,6 +703,17 @@ irqstate_t up_irq_enable(void);
 int up_cpu_index(void) noinstrument_function;
 #endif /* CONFIG_ARCH_HAVE_MULTICPU */
 
+/****************************************************************************
+ * Schedule acceleration macros
+ ****************************************************************************/
+
+#ifdef CONFIG_RISCV_PERCPU_SCRATCH
+#define up_current_regs()      (this_task()->xcp.regs)
+#define up_this_task()         (*((struct tcb_s **)READ_CSR(CSR_SCRATCH)))
```

Comment:
@pussuw Hello, I noticed that the origin definition of `riscv_percpu_s` is from you, I think it's better to move `riscv_percpu_s` to a more public place then we can access it here, do you have any suggestions?

Reply:
Or maybe we should use TP register directly instead

Reply:
How much more public did you have in mind ?

Reply:
I guess it should be same level with https://github.com/apache/nuttx/pull/15081/files#diff-3a3de6a3c30afaa1d7fd006bb5562ed90080afb6b3a3c8202c06835ccd6835d4

Reply:
Yeah I guess arch.h would be the place then.

Did you test if this is faster than fetching the readytorun head ? I would presume reading the memory mapped CSR is just as slow.

Reply:
I haven't done a formal test yet. In theory,
```c
#define up_this_task() (*((struct tcb_s **)READ_CSR(CSR_SCRATCH)))
```
should be more faster than:
https://github.com/apache/nuttx/blob/0e1b432dd068880d353ae91c01e610f85b4a16ea/sched/sched/sched.h#L367-L387
(It would be faster in an SMP scenario)